pkg/kubelet/cm/dra/plugin/plugin.go

```diff
@@ -17,22 +17,69 @@ limitations under the License.
 package plugin
 
 import (
+	"context"
 	"errors"
 	"fmt"
+	"net"
 	"strings"
+	"sync"
+	"time"
 
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/connectivity"
+	"google.golang.org/grpc/credentials/insecure"
 	utilversion "k8s.io/apimachinery/pkg/util/version"
 	"k8s.io/klog/v2"
 )
 
 const (
 	// DRAPluginName is the name of the in-tree DRA Plugin.
-	DRAPluginName = "kubernetes.io/dra"
+	DRAPluginName   = "kubernetes.io/dra"
+	v1alpha3Version = "v1alpha3"
+	v1alpha2Version = "v1alpha2"
 )
 
-// draPlugins map keeps track of all registered DRA plugins on the node
-// and their corresponding sockets.
-var draPlugins = &PluginsStore{}
+// Plugin is a description of a DRA Plugin, defined by an endpoint
+// and the highest DRA version supported.
+type plugin struct {
+	sync.RWMutex
```

Comment:
Mutex is still here. Is it really needed?

Reply:
Yes it is used when retrieving or creating the gRPC conn and also when setting the actual plugin version supported

Reply:
Yes, but is it needed? I thought that we don't call GRPC asynchronously, do we?

Reply:
I wasn't sure about calling GRPC synchronously every time so I kept it. It is removed in [this](535d058166a) commit.

Reply:
Let's see how that influences unit and e2e tests. If all tests pass then we don't need this mutex.

Reply:
Looks like all tests passed. If you can increase a code coverage in this are it would be great! otherwise lgtm.

Reply:
Yes, it seems we don't need the mutex. good news. I'm already looking into code coverage. I will update the PR asap.